packages/runtime/src/store/previews.ts

```diff
@@ -57,18 +59,15 @@ export class PreviewsStore {
 
     const previewInfos = previews.map((preview) => {
       const info = new PreviewInfo(preview);
+      const portInfo = this._availablePreviews.get(info.port);
 
-      let previewInfo = this._availablePreviews.get(info.port);
-
-      if (!previewInfo) {
-        previewInfo = info;
-
-        this._availablePreviews.set(previewInfo.port, previewInfo);
+      if (!portInfo) {
+        this._availablePreviews.set(info.port, info.portInfo);
       } else {
-        previewInfo.title = info.title;
+        info.portInfo = portInfo;
```

Comment:
When webcontainer's `port` event is emitted, we mark a single `PortInfo` as ready. This single `PortInfo` can be referenced by multiple `PreviewInfo`'s. That's why need to be able to explicitly set `info.portInfo` here. 

@Nemikolh this is not exactly in line with your first review - let me know what you think!